src/axis.js

```diff
@@ -81,13 +83,19 @@ function axis(orient, scale) {
           .attr("opacity", epsilon)
           .attr("transform", function(d) { return isFinite(d = position(d)) ? transform(d + offset) : this.getAttribute("transform"); });
 
+      path.attr("opacity", 1);
+      pathExit = pathExit.transition(context)
+          .attr("opacity", epsilon);
```

Comment:
Could use stroke-opacity here for performance.

Reply:
on ticks also?

Reply:
No, because ticks also have text.